### PR TITLE
feat: Add delete file utility

### LINKED ISSUE #145: Delete file
**Dev story**
Write a function/method that deletes a file (equiv to Linux command `rm`)

Example:
```go
err := rm(filepath)
```

**Acceptance Criteria**
- [x] Make sure it works cross-platform
- [x] Put the function into `internal/utils` package
- [x] For logging use https://github.com/Sirupsen/logrus
- [x] Write a unit test
- [x] Make sure there are [testable examples](https://blog.golang.org/examples)

**Hints**:
* Writing a unit test for filesystem checking is not trivial. You should NOT create a real file on the system, because then your test will depend on the I/O of the file system itself. The last resort is mocking the filesystem. There are quite a few powerful libraries like spf13/afero for this purpose (mocking of a filesystem). These packages will create temporary files in the background and clean up afterward.
* As an exercise, try to write the test first. Let it fail. Then try to fix it, by writing the actual implementation of the function.

**Extra Help**:
* https://golangcode.com/check-if-a-file-exists/
* https://www.youtube.com/watch?v=ifBUfIb7kdo
* https://github.com/spf13/afero#using-afero-for-testing
* An example using Afero for testing: http://krsacme.com/golang-fs-tests/
* For logging https://linuxhint.com/golang-logrus-package/

**NOTICE**: If you are unable to mock this test, in worst case scenario, create a folder `internal/utils/testdata` and put actual files you can use for your test.

### PR DESCRIPTION
**Pull request by ge0r**
Fixes #145

### CODE REVIEW

File: internal/utils/filesystem.go [hunk 1]
```diff
@@ -32,3 +32,17 @@ func FolderExists(path string) bool {
 func FileExists(path string) bool {
 	return exists(path, false)
 }
+
+// Remove deletes the specified file
+func Remove(path string) error {
+	if path == "" {
+		log.Debug("Path is empty")
```
Comment (zoispag) [inline internal/utils/filesystem.go]: Since we are returning an error, let's drop the log line.

**Review by zoispag — COMMENTED** [review]
LGTM.
Q: Are we taking permissions into account?

**Comment by ge0r** [comment]
> LGTM.
> Q: Are we taking permissions into account?

`os.Remove(path)` does, and I m just returning it's error.
Do you think I should have a specific test case for it either way?

**Comment by zoispag** [comment]
> > LGTM.
> > Q: Are we taking permissions into account?
> 
> `os.Remove(path)` does, and I m just returning it's error.
> Do you think I should have a specific test case for it either way?

If it's not too much trouble, I'd say yes.

**Comment by zoispag** [comment]
@Mergifyio update

**Comment by ge0r** [comment]
@Mergifyio update

**Comment by ge0r** [comment]
> > > LGTM.
> > > Q: Are we taking permissions into account?
> > 
> > 
> > `os.Remove(path)` does, and I m just returning it's error.
> > Do you think I should have a specific test case for it either way?
> 
> If it's not too much trouble, I'd say yes.

So, after all, I m just creating an immutable file which does not allow `Remove()` to delete it, and I added a test case to check that.

File: internal/utils/filesystem_test.go [hunk 2]
```diff
@@ -121,3 +136,75 @@ func TestFileExists(t *testing.T) {
...
+	} else {
+		fmt.Println("Unable to delete file")
+	}
+	// Output: File deleted
+}
+
+func removeCleanup() {
+	err := chattr.RemoveImmutable(fileNotToDelete)
+	if err != nil {
+		fmt.Println(err)
```
Comment (zoispag) [inline internal/utils/filesystem_test.go]:
```suggestion
        log.Fatalf("Coult not remove immutable file %s", err)
```

File: internal/utils/filesystem_test.go [hunk 3]
```diff
@@ -121,3 +136,75 @@ func TestFileExists(t *testing.T) {
...
+	}
+}
+func TestRemove(t *testing.T) {
+	createFile(fileToDelete)
+
+	err := chattr.AddImmutable(fileNotToDelete)
+
+	if err != nil {
+		fmt.Println(err)
```
Comment (zoispag) [inline internal/utils/filesystem_test.go]:
```suggestion
        log.Fatalf("Coult not create immutable file %s", err)
```

File: internal/utils/filesystem_test.go [hunk 4]
```diff
@@ -121,3 +136,75 @@ func TestFileExists(t *testing.T) {
...
+	// Output: File deleted
+}
+
+func removeCleanup() {
+	err := chattr.RemoveImmutable(fileNotToDelete)
+	if err != nil {
+		fmt.Println(err)
+	}
+}
+func TestRemove(t *testing.T) {
```
Comment (zoispag) [inline internal/utils/filesystem_test.go]:
```suggestion
}

func TestRemove(t *testing.T) {
```

**Review by zoispag — APPROVED** [review]

**Review by drpaneas — APPROVED** [review]

**Pull request merged**